### PR TITLE
List searching

### PR DESCRIPTION
**Pull request by jakedaleweb**

### CODE REVIEW

File: lib/slack_interface/request_handler.coffee [hunk 1]
```diff
@@ -112,7 +112,7 @@ class SlackInterfaceRequestHandler
                   reply_data['text'] = "And we're back. Hope you closed that deal!"
 
               when 'list'
-                if @auth.args[0]?
+                if @auth.args[0]? && @auth.args[0] != '|'
```
Comment (andrewandante) [inline lib/slack_interface/request_handler.coffee]: Probably switch these - start with `if @auth.args[0] == '|'`, and then `else if @auth.args[0]?`. Keeps it clean.

File: lib/slack_interface/request_handler.coffee [hunk 2]
```diff
@@ -127,6 +127,14 @@ class SlackInterfaceRequestHandler
                     reply_data['text'] = ['Ok.', 'Sweet.', 'Chur.', 'Done like dinner.', 'sorted.org.nz (use your mouse!)', 'Coolies.', 'No problem, brah.', 'Affirmative.', 'Gotcha.', 'Aye-aye, captain! :captain:'][Math.floor(Math.random() * 10)]
                   else
                     reply_data['text'] = "Oops, you did it again. Try `help` if you need some."
+                else if @auth.args[0] == '|' and @auth.args[1] == 'grep'
```
Comment (andrewandante) [inline lib/slack_interface/request_handler.coffee]: I'd remove the need for `grep` - introduces some weird edge cases where you could do `list | notgrep` etc. While I appreciate turning it into a terminal, I think either `list | <searchterm>` or even just `search <searchterm>` or `searchlist <searchterm>` might be better here.

There's some argument for `list search <searchterm>` too - you'd just have to blacklist it as a playlist name.

File: lib/slack_interface/request_handler.coffee (hunk 1 above)
Comment (jakedaleweb) [reply lib/slack_interface/request_handler.coffee]: agree

File: lib/slack_interface/request_handler.coffee (hunk 2 above)
Comment (jakedaleweb) [reply lib/slack_interface/request_handler.coffee]: disagree, no plebs allowed. What if we do both? fair compromise (:

Comment (NightJar) [reply lib/slack_interface/request_handler.coffee]: Call it `find`. WTB; `search` for 'grepping' Spotify proper.

**Pull request merged**